.changeset/new-jobs-deny.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog-react': minor
+---
+
+Add possibility to re-use EntityPicker for filters with multiple select.
+**BREAKING** Style name `CatalogReactEntityTagPicker` was changed to `EntityAutocompletePickerInput`
```

Comment:
OK I'm having second thoughts about this one.

Check [this out](https://github.com/backstage/backstage/blob/6ab9bfd301aca4b91be74e83aa9e0f79ffca8321/plugins/catalog-react/src/overridableComponents.ts#L29). That's a public interface collecting the theme-overridable things.

First thing to note: They are ALL prefixed `CatalogReact` after the name of the package they are in, to avoid collisions. This is important because this all gets part of the global space of things to override in the theme. So IF we make new things we need to have that prefix.

But:

The second thing is, note that the public API surface isn't changed (in the api-report). End users will be surprised that the component isn't changed, but the style is called something unrelated. So I think we should revert this after all.

In the future, if we make this a public thing, I think we need to make the styling split into pieces. For example, maybe we would have a style for `.CatalogReactEntityTagPicker-input`, that's _passed down_ to the underlying input as a prop? Rather than styling the underlying thing separately. But for now let's not worry about it.

Reply:
so we will keep `CatalogReactEntityTagPicker` naming for generic `autocomplete` picker ? can we just add `CatalogReact` at least `CatalogReactEntityAutocompletePickerInput`, maybe a bit long though, but at least something more relevant than `tag picker`

Reply:
Think about it from an end user perspective.

I was using component `Foo`. I styled it with the style name `CatalogReactFoo`. Now somebody does a completely hidden internal refactor of `Foo` and then the style name changes to the name of that internal thing, instead of the public thing I'm supposedly styling.

I think what should happen from an end user perspective, if `Foo` now has a new edit box in it, then I should have the new _ability_ (but not requirement) to style also `CatalogReactFoo-input` for example.

Do you see some way that that could be achieved? Maybe `Foo` does `useStyles`, and then GIVES the class names down to the implementation component underneath, without the end user knowing that that's what happens? The inner component would have a prop `classes: { input?: string }`, which [seems to be the pattern](https://github.com/backstage/backstage/blob/8ba579e8bd47793492e3e8f0b8c8077399a7bb45/packages/core-components/src/components/OAuthRequestDialog/OAuthRequestDialog.tsx#L75) they use in MUI in general.

Reply:
ok, so what do you think about such solution: we will rename to `CatalogReactEntityAutocompletePickerInput` so will keep consistency, also we can pass `CatalogReactEntityTagPicker-input` to input, so backwards compatibility will preserve